jobs/ml_jobs/algo_training/two_towers_model/models/two_towers_model.py

```diff
@@ -16,61 +16,53 @@ def __init__(
         embedding_size: int,
     ):
-        """
-        data: DataFrame of user-item interactions along with their corresponding features
-        user_features_config: dict containing the information about the user input embedding layers
-        item_features_config: dict containing the information about the item input embedding layers
-        items_dataset: TF dataset containing the features for each item
-        embedding_size: Final embedding size for the items & the users
-        """
         super().__init__()
-
-        # dict of preprocessing layers for each user feature
-        user_embedding_layers = {
-            name: self.load_embedding_layer(
-                layer_type=layer["type"],
-                embedding_size=layer["embedding_size"],
-            )
-            for name, layer in user_features_config.items()
-        }
-        self._user_feature_names = user_features_config.keys()
-
-        # dict of preprocessing layers for each item feature
-        item_embedding_layers = {
-            name: self.load_embedding_layer(
-                layer_type=layer["type"],
-                embedding_size=layer["embedding_size"],
-                embedding_initialisation_weights=data[name].drop_duplicates().tolist()
-                if layer["type"] == "pretrained"
-                else None,
-            )
-            for name, layer in item_features_config.items()
-        }
-
         self._item_feature_names = item_features_config.keys()
+        self._user_feature_names = user_features_config.keys()
 
+        # Define user and item models
         self.user_model = SingleTowerModel(
             data=data,
-            input_embedding_layers=user_embedding_layers,
+            input_embedding_layers={
+                name: self.load_embedding_layer(
+                    layer_type=layer["type"],
+                    embedding_size=layer["embedding_size"],
+                )
+                for name, layer in user_features_config.items()
+            },
             embedding_size=embedding_size,
         )
+
         self.item_model = SingleTowerModel(
             data=data,
-            input_embedding_layers=item_embedding_layers,
+            input_embedding_layers={
+                name: self.load_embedding_layer(
+                    layer_type=layer["type"],
+                    embedding_size=layer["embedding_size"],
+                    embedding_initialisation_weights=data[name]
+                    .drop_duplicates()
+                    .tolist()
+                    if layer["type"] == "pretrained"
+                    else None,
+                )
+                for name, layer in item_features_config.items()
+            },
             embedding_size=embedding_size,
         )
 
+        item_embeddings = item_dataset.map(self.item_model)
+        index_top_k = tfrs.layers.factorized_top_k.BruteForce()
+        index_top_k.index_from_dataset(item_embeddings)
```

Comment:
il ne manque pas un index_top_k = ? 

parce que soit dans ton code, cela signifie que index_top_k est in place et dans ce cas là, on s'attend à ce que l'ancien code renvoie none 
```
tfrs.layers.factorized_top_k.BruteForce().index_from_dataset(
                    items_dataset.map(self.item_model)
                )
```,
soit on ne fait rien du return

Reply:
Dans les deux cas il retourne self !